main.py

```diff
@@ -176,30 +209,68 @@ def clicked(self, position):
         for ball in self.balls:
             # Check if the mouse position is inside the surface of the ball
             if (distance(position, ball.position) < BALL_RADIUS):
-                # Case(1): No ball is selected atm and the observed ball is not white.
-                #   -> Select it and set depending variables accordingly.
-                if not self.isBallSelected and ball.colour is not WHITE:
-                    ball.select()
-                    self.selectedBall = ball
-                    self.isBallSelected = True
-                # Case(2): A ball is selected atm and the observed ball has a big rim.
-                #   -> Unselect all balls and set depending variables accordingly.
-                elif self.isBallSelected and ball.selected == BALL_MARK_L:
-                    for ball in self.balls:
-                        ball.select(False)
-                    self.isBallSelected = False
-                # Case(3): A ball is selected atm and the observed ball has a small rim.
-                #   -> Swap the colours of the selected and the observed ball, unselect
-                #      all balls and set depending variables accordingly.
-                elif self.isBallSelected and ball.selected == BALL_MARK_S:
-                    ball.colour = self.selectedBall.colour
-                    self.selectedBall.colour = WHITE
-                    for ball in self.balls:
-                        ball.select(False)
-                    self.isBallSelected = False
-                    self.counter += 1
+                self.move(ball)
                 break
 
+    # add comments
+    def setState(self, state):
+        for i in range(len(self.balls)):
+            self.balls[i].colour = state[i]
+            self.balls[i].selected = BALL_NOMARK
+            self.isBallSelected = False
+
+    # add comments
+    def getState(self):
+        state = []
+        for i in range(len(self.balls)):
+            state.append(self.balls[i].colour)
+        return state
+
+# add comments
+class State:
```

Comment:
Class desctiption is missing

Reply:
added description

Reply:
Can't find it.
Can be shifted to a later merge.
```suggestion
class State:
"""
The class State: 
TODO[ADD] Description
"""
```

Reply:
added "TODO" as suggested